frontend/src/views/HomeView.vue

```diff
@@ -1 +1,68 @@
-<template><h1>Test</h1></template>
+<template>
+    <div class="mb-4">
+      <ElButton
+        key="primary"
+        type="text"
+        @click="showConfirmationDialog(roleId)"
+        style="color: #29517A;"
+      >
+        Delete
+      </ElButton>
+    </div>
+    <ElDialog
+      v-model="dialogVisible"
+      title="Warning"
+      width="500"
+    >
+      <span>Role will be permanently removed. Continue?</span>
+  
+      <template #footer>
+        <div class="dialog-footer">
+          <ElButton @click="dialogVisible = false">Cancel</ElButton>
+          <ElButton type="primary" @click="deleteRole()">
+            Confirm
+          </ElButton>
+        </div>
+      </template>
+    </ElDialog>
+  </template>
+  
+  <script setup lang="ts">
+  import { ElButton, ElDialog, ElMessage } from 'element-plus';
+  import { ref } from 'vue'
+  
+  const dialogVisible = ref(false)
+  const roleId = ref<string>('')
+  
+  const showConfirmationDialog = (id: string) => {
+    dialogVisible.value = true;
+    roleId.value = id; 
+  }
+  
+  const deleteRole = () => {
+  fetch(`/roles/${parseInt(roleId.value)}`, {
```

Comment:
Please don't call this directly, to separate the responsibilities of the files.
Take a look at how we fetch the roles in HomeView.ts through rolesStore.fetchRoles(), and try to follow the pattern.